keras_cv/models/object_detection/yolo_v8/yolo_v8_detector.py

```diff
@@ -0,0 +1,632 @@
+        ]
+        feature_extractor = get_feature_extractor(
+            backbone, extractor_layer_names, extractor_levels
+        )
+
+        images = layers.Input(feature_extractor.input_shape[1:])
+        features = list(feature_extractor(images).values())
+
+        fpn_features = apply_path_aggregation_fpn(
+            features, depth=fpn_depth, name="pa_fpn"
+        )
+
+        outputs = apply_yolov8_head(
+            fpn_features,
+            num_classes,
+        )
+
+        # To make loss metrics pretty, we use a no-op layer with a good name.
+        boxes = keras.layers.Concatenate(axis=1, name="box")([outputs["boxes"]])
+        scores = keras.layers.Concatenate(axis=1, name="class")(
+            [outputs["classes"]]
+        )
+
+        outputs = {"boxes": boxes, "classes": scores}
+        super().__init__(inputs=images, outputs=outputs, **kwargs)
+
+        self.bounding_box_format = bounding_box_format
+        self.prediction_decoder = (
+            prediction_decoder
+            or keras_cv.layers.MultiClassNonMaxSuppression(
+                bounding_box_format=bounding_box_format,
+                from_logits=False,
+                confidence_threshold=0.2,
+                iou_threshold=0.7,
+            )
+        )
+        self.backbone = backbone
+        self.fpn_depth = fpn_depth
+        self.num_classes = num_classes
+        self.label_encoder = label_encoder or YOLOV8LabelEncoder(
+            num_classes=num_classes
+        )
+
+    def compile(
+        self,
+        box_loss,
+        classification_loss,
+        box_loss_weight=7.5,
+        classification_loss_weight=0.5,
+        metrics=None,
+        **kwargs,
+    ):
+        """Compiles the YOLOV8Detector.
+
+        `compile()` mirrors the standard Keras `compile()` method, but has one
+        key distinction -- two losses must be provided: `box_loss` and
+        `classification_loss`.
+
+        Args:
+            box_loss: a Keras loss to use for box offset regression. A
+                preconfigured loss is provided when the string "iou" is passed.
+            classification_loss: a Keras loss to use for box classification. A
+                preconfigured loss is provided when the string
+                "binary_crossentropy" is passed.
+            box_loss_weight: (optional) float, a scaling factor for the box
+                loss. Defaults to 7.5.
+            classification_loss_weight: (optional) float, a scaling factor for
+                the classification loss. Defaults to 0.5.
+            kwargs: most other `keras.Model.compile()` arguments are supported
+                and propagated to the `keras.Model` class.
+        """
+        if metrics is not None:
+            raise ValueError("User metrics not yet supported for YOLOV8")
+
+        if isinstance(box_loss, str):
+            if box_loss == "iou":
+                box_loss = YOLOV8IoULoss(reduction="sum")
+            else:
```

Comment:
Couldn't other losses be provided by string like `"mean_squared_error"`?
🤔 🤔 🤔

Reply:
Yes, although these are really the only sensible defaults. In my view, supporting it as a string is a way of saying "this is an appropriate loss to use here, but we want you to pass it as a string to make it explicit"